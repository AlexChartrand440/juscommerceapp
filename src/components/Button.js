import React from 'react';
import styled from 'styled-components';
import { Button } from 'react-native-paper';

const SelectButton = styled(Button)`
  width: 250;
  height: 45;
  background-color: #ff685f;
  border-radius: 5;
  justify-content: center;
  align-items: center;
  align-self: center;
`;

const SelectButtonText = styled.Text`
  width: 100%;
  color: #fff;
  font-size: 16;
  font-weight: 300;
`;

export default ({ label, onPress }) => (
  <SelectButton onPress={onPress}>
    <SelectButtonText>{label}</SelectButtonText>
  </SelectButton>
);
