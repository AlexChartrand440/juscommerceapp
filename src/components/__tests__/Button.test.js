import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../Button';

test('Button', () => {
  const wrapper = renderer.create(<Button />).toJSON();
  expect(wrapper).toMatchSnapshot();
});
