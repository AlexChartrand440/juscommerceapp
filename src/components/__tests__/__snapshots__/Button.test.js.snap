// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "center",
      "backgroundColor": "#ff685f",
      "borderColor": "transparent",
      "borderRadius": 5,
      "borderStyle": "solid",
      "borderWidth": 0,
      "elevation": 0,
      "height": 45,
      "justifyContent": "center",
      "minWidth": 64,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 0.24,
      "shadowRadius": 0,
      "width": 250,
    }
  }
>
  <View
    accessibilityRole="button"
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 4,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          },
          undefined,
        ]
      }
    >
      <Text
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "Helvetica Neue",
              "textAlign": "left",
              "writingDirection": "ltr",
            },
            Array [
              Object {
                "letterSpacing": 1,
                "marginHorizontal": 16,
                "marginVertical": 9,
                "textAlign": "center",
              },
              undefined,
              Object {
                "color": "#6200ee",
                "fontFamily": "HelveticaNeue-Medium",
              },
              Object {
                "fontFamily": "HelveticaNeue-Medium",
              },
            ],
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 16,
                "fontWeight": "300",
                "width": "100%",
              },
            ]
          }
        />
      </Text>
    </View>
  </View>
</View>
`;
